include/drivers/adc.h

```diff
@@ -473,6 +551,48 @@ static inline uint16_t adc_ref_internal(const struct device *dev)
 	return api->ref_internal;
 }
 
+__syscall int adc_config_threshold_irq(const struct device *dev,
```

Comment:
Need doxygen tags describing the function

Reply:
Comments added.